patchwork-events-entity/src/main/java/net/patchworkmc/mixin/event/entity/MixinServerPlayerEntity.java

```diff
@@ -49,4 +58,52 @@ private void hookCopyFromForCloneEvent(ServerPlayerEntity oldPlayer, boolean ali
 		ServerPlayerEntity speThis = (ServerPlayerEntity) (Object) this;
 		MinecraftForge.EVENT_BUS.post(new PlayerEvent.Clone(speThis, oldPlayer, !alive));
 	}
+
+	////////////////////////////////////
+	// PlayerChangedDimensionEvent
+	////////////////////////////////////
+	@Inject(method = "teleport",
+			at = @At(
+					value = "INVOKE",
+					shift = Shift.AFTER,
+					ordinal = 0,
+					target = "net/minecraft/server/PlayerManager.method_14594(Lnet/minecraft/server/network/ServerPlayerEntity;)V"
+					),
+			locals = LocalCapture.CAPTURE_FAILHARD
+	) // PlayerManager.method_14594 -> sendInventory
+	private void teleport_sendInventory(ServerWorld targetWorld, double x, double y, double z, float yaw, float pitch, CallbackInfo info, ServerWorld serverWorld, LevelProperties levelProperties) {
+		ServerPlayerEntity me = (ServerPlayerEntity) (Object) this;
+		PlayerEvents.firePlayerChangedDimensionEvent(me, serverWorld.dimension.getType(), me.dimension);
+	}
+
+	@Unique
+	private static final ThreadLocal<DimensionType> changeDimension_from = new ThreadLocal<>();
```

Comment:
Does this need to be a thread local?

Reply:
In case the same method is called at the same time on different threads, especially since it's being stored as a static variable instead of a variable on the class itself, i believe.